kayenta-judge/src/main/scala/com/netflix/kayenta/judge/classifiers/metric/MannWhitneyClassifier.scala

```diff
@@ -55,6 +57,15 @@ class MannWhitneyClassifier(tolerance: Double=0.25, confLevel: Double=0.95) exte
     (lowerBound, upperBound)
   }
 
+  /**
+    * Add Gaussian noise to the input array
+    */
+  private def addGaussianNoise(values: Array[Double]): Array[Double] = {
+    val scalingFactor = 1e-5
+    val metricScale = values.distinct.head * scalingFactor
```

Comment:
what happens if the first value is 0 or NaN?

Reply:
If this happens then the `metricScale` would be 0 and NaN respectively. Note that, NaNs are transformed, either removed or replaced, before the classifier is called. In spite of this, I will add some guards to check for the zero or NaN case.